firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift

```diff
@@ -0,0 +1,902 @@
+        }
+    }
+}
+
+protocol TrackingProtectionMenuDelegate: AnyObject {
+    func settingsOpenPage(settings: Route.SettingsSection)
+    func didFinish()
+}
+
+class TrackingProtectionViewController: UIViewController, Themeable, Notifiable, UIScrollViewDelegate, BottomSheetChild {
+    var themeManager: ThemeManager
+    var themeObserver: NSObjectProtocol?
+    var notificationCenter: NotificationProtocol
+    let windowUUID: WindowUUID
+    var currentWindowUUID: UUID? { windowUUID }
+
+    weak var enhancedTrackingProtectionMenuDelegate: TrackingProtectionMenuDelegate?
+
+    private var faviconHeightConstraint: NSLayoutConstraint?
+    private var faviconWidthConstraint: NSLayoutConstraint?
+    private var foxImageHeightConstraint: NSLayoutConstraint?
+    private var shieldImageHeightConstraint: NSLayoutConstraint?
+    private var lockImageHeightConstraint: NSLayoutConstraint?
+
+    private var trackersArrowHeightConstraint: NSLayoutConstraint?
+    private var connectionArrowHeightConstraint: NSLayoutConstraint?
+
+    private lazy var scrollView: UIScrollView = .build { scrollView in
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.showsVerticalScrollIndicator = false
+    }
+
+    private let baseView: UIView = .build { view in
+        view.translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    // MARK: UI components Header View
+    private let headerContainer: UIView = .build { view in
+        view.backgroundColor = .clear
+    }
+
+    private lazy var headerLabelsContainer: UIStackView = .build { stack in
+        stack.backgroundColor = .clear
+        stack.alignment = .leading
+        stack.axis = .vertical
+        stack.spacing = TPMenuUX.UX.headerLabelDistance
+    }
+
+    private var favicon: FaviconImageView = .build { favicon in
+        favicon.manuallySetImage(
+            UIImage(named: StandardImageIdentifiers.Large.globe)?.withRenderingMode(.alwaysTemplate) ?? UIImage())
+    }
+
+    private let siteDisplayTitleLabel: UILabel = .build { label in
+        label.font = FXFontStyles.Regular.headline.scaledFont()
+        label.numberOfLines = TPMenuUX.UX.headerLinesLimit
+        label.adjustsFontForContentSizeCategory = true
+        label.translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    private let siteDomainLabel: UILabel = .build { label in
+        label.font = FXFontStyles.Regular.caption1.scaledFont()
+        label.numberOfLines = 0
+        label.adjustsFontForContentSizeCategory = true
+        label.translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    private var closeButton: UIButton = .build { button in
+        button.layer.cornerRadius = 0.5 * TPMenuUX.UX.closeButtonSize
+        button.clipsToBounds = true
+        button.imageView?.contentMode = .scaleAspectFit
+        button.setImage(UIImage(named: StandardImageIdentifiers.Medium.cross), for: .normal)
+    }
+
+    // MARK: Connection Details View
```

Comment:
could we separate this UI components in its own view and then instantiate them in this view controller. It can avoid this VC become too massive and decouple the views more.

Reply:
Since the views are different one from each other, do you have a suggestion how this "decouple" should looks like? Thank you.

Reply:
I was thinking something along the line of how we have `MicrosurveyConfirmationView` and then its being used in `MicrosurveyViewController`. In this case, we can create a separate `TrackingProtectionConnectionDetailsView` and then call it in the view controller here.

Something like this, let me know if this makes sense. Happy to chat about it as well to clarify!
```
private lazy var connectionDetailsView: TrackingProtectionConnectionDetailsView = .build()
```

Reply:
but my suggestion: in order to not hold this PR only for a refactor, let me to create a ticket and put it as a TODO in the TrackingProtectionViewController file and it can be handled in future.